src/main/java/net/sf/mzmine/modules/peaklistmethods/alignment/adap3/ADAP3AlignerTask.java

```diff
@@ -0,0 +1,419 @@
+                        Double.toString(intensity));
+            }
+
+            newRow.setPeakInformation(information);
+
+            alignedPeakList.addRow(newRow);
+        }
+
+        return alignedPeakList;
+    }
+
+    private Component getComponent(final PeakListRow row) {
+        final PeakIdentity identity = row.getPreferredPeakIdentity();
+
+        if (row.getNumberOfPeaks() == 0)
+            throw new IllegalArgumentException("No peaks found");
+
+        // Read Spectrum information        
+        NavigableMap<Double, Double> spectrum = new TreeMap<>();
+
+        IsotopePattern pattern = row.getBestIsotopePattern();
+
+        if (pattern == null)
+            throw new IllegalArgumentException("ADAP Alignment requires mass "
+                    + "spectra (or isotopic patterns) of peaks. No spectra found.");
+
+        for (DataPoint dataPoint : pattern.getDataPoints())
+            spectrum.put(dataPoint.getMZ(), dataPoint.getIntensity());
+
+        // Read Chromatogram
+        final Feature peak = row.getBestPeak();
+        final RawDataFile dataFile = peak.getDataFile();
+
+        NavigableMap<Double, Double> chromatogram = new TreeMap<>();
+
+        for (final int scan : peak.getScanNumbers()) {
+            final DataPoint dataPoint = peak.getDataPoint(scan);
+            if (dataPoint != null)
+                chromatogram.put(dataFile.getScan(scan).getRetentionTime(),
+                        dataPoint.getIntensity());
+        }
+
+        return new Component(null,
+                new Peak(chromatogram, new PeakInfo()
+                        .mzValue(peak.getMZ())
+                        .peakID(row.hashCode())),
+                spectrum, null);
+    }
+
+    private void process(Project alignment) {
+        AlignmentParameters params = new AlignmentParameters()
+                .sampleCountRatio(parameters.getParameter(
+                        ADAP3AlignerParameters.SAMPLE_COUNT_RATIO).getValue())
+                .retTimeRange(parameters.getParameter(
+                        ADAP3AlignerParameters.RET_TIME_RANGE).getValue().getTolerance())
+                .scoreTolerance(parameters.getParameter(
+                        ADAP3AlignerParameters.SCORE_TOLERANCE).getValue())
+                .scoreWeight(parameters.getParameter(
+                        ADAP3AlignerParameters.SCORE_WEIGHT).getValue())
+                .maxShift(2 * parameters.getParameter(
+                        ADAP3AlignerParameters.RET_TIME_RANGE).getValue().getTolerance())
+                .eicScore(parameters.getParameter(
+                        ADAP3AlignerParameters.EIC_SCORE).getValue());
+
+        params.optimizationParameters = new OptimizationParameters()
+                .gradientTolerance(1e-6)
+                .alpha(1e-4)
+                .maxIterationCount(4000)
+                .verbose(false);
+
+        alignment.alignSamples(params);
+    }
+
+    private PeakListRow findRow(final PeakList[] peakLists,
```

Comment:
the parameter peakLists is redundant, because you already have a private field peakLists